numpy/core/fromnumeric.py

```diff
@@ -1399,9 +1399,9 @@ def resize(a, new_shape):
     """
     if isinstance(new_shape, (int, nt.integer)):
         new_shape = (new_shape,)
```

Comment:
A counter-proposal, @seberg - what if we implemented this as:
```python
ret = np.empty_like(a, shape=new_shape, order='C').reshape((-1,))
... # as before
a = concatenate((a,) * n_copies, out=ret)
... # as before
```

Then all the error handling gets done by the ndarray constructor.

Reply:
Yeah, does seem a bit cleaner. Enforcing C order is probably also pretty much what it was before (not quite sure if concatenate does this). Maybe even shorter/cleaner (although a bit more tricky):
```
ret = np.empty_like(a, shape=new_shape, order="C")
ret.flat[:] = a
return ret
```

Reply:
I'd forgotten `flat` had those super weird semantics.

That doesn't do the right thing when `len(a)` == 0 though